### DISCUSSION TITLE
Need to access the routeParam on a route page.

### DISCUSSION THREAD

**Discussion by m-tranter** [vikejs/vike#1051] [Discussion] [answered]
I have some code like this: 
`  computed: {
    item() {
      return this.$store.state.copyItems.find((x) => x.sys.slug === pageContext.routeParams.event);
    }
  },
`
Obviously, this doesn't work. I have been reading the documentation, but I can't understand how to have access to that param on my route page - I need it to fetch something from the store.

**Comment by m-tranter** [comment]
I've tried some other approaches - such as setting the item with my link, but that didn't work either.
  <a  @click="setItem(item)" :href="`events/${item.sys.slug}`"


methods: {
    setItem: function (obj) {
      this.$store.dispatch('setItem', obj);
    },

setItem({ commit }, obj) {
        commit('SET_ITEM', obj);
      },

SET_ITEM(state, obj) {
        state.item = obj;
      },

**Comment by m-tranter** [comment]
I just want to navigate to the page with the slug of this item / object as the url and have access to the item in order to use its properties in the template.

**Comment by brillout** [comment]
>  how to have access to that param

Have you seen https://vite-plugin-ssr.com/pageContext-anywhere?

**Reply by m-tranter** [reply]
Thanks. I have solved that but now I can't access the store in this route page.

**Reply by brillout** [reply, marked as answer]
Seems more like a Vue question than a VPS one.

**Comment by m-tranter** [comment]
Probably. But I don't understand. I'm getting access to the store in components, but not here. Do I need to use a component on the route page?

**Reply by brillout** [reply]
I'm not that familiar with Vue, I'd ask in Vue forums instead.
